src/common/src/lru.rs

```diff
@@ -323,7 +352,7 @@ where
         }
     }
 
-    fn detach(&mut self, mut ptr: NonNull<LruEntry<K, V>>) {
+    fn detach(mut ptr: NonNull<LruEntry<K, V>>) {
```

Comment:
I'm not entirely sure but previously `&mut self` ensured that caller has a mutable reference to the cache. Will removing it somehow affects optimization and leads to unsoundness?

Reply:
During all the calls to this method, the caller `self` is held as a mut ref, so I think nothing is changed.